[Beta] docs(cn): translate learn/index into Chinese (#696)
reactjs/zh-hans.reactjs.org: beta/src/pages/learn/index.md

---
title: 开始

> ---

Reviewer: ```suggestion
translator:
 - x-cold
 - sooia（刘俊）
```

Reviewer: 添加翻译者和Review者名字。

<Intro>

欢迎阅读 React 文档! 此文档是您在本站点能找到的信息的概览。

</Intro>

<YouWillLearn>

* [如何安装 React](/learn/installation)
* [如何用 React 的思考方式构建用户界面](/learn/thinking-in-react)
* [如何在屏幕中渲染内容](/learn/describing-the-ui)
* [如何让您的用户界面响应交互](/learn/adding-interactivity)
* [随着您的应用的增长，如何保持逻辑的可维护性](/learn/managing-state)

> * [必要时如何使用安全舱逃离 React](/learn/escape-hatches)

Reviewer: ```suggestion
* [必要时如何退出 React](/learn/escape-hatches)
```

* [How to keep your logic maintainable as your app grows](/learn/managing-state)
* [How to step outside React when necessary](/learn/escape-hatches)
* [如何安装 React](/learn/installation)
* [如何用 React 的思考方式构建用户界面](/learn/thinking-in-react)
* [如何在屏幕中渲染内容](/learn/describing-the-ui)
* [如何让您的用户界面响应交互](/learn/adding-interactivity)
* [随着您的应用的增长，如何保持逻辑的可维护性](/learn/managing-state)
* [必要时如何使用安全舱逃离 React](/learn/escape-hatches)

</YouWillLearn>

> ## Introduction {/*introduction*/}

Reviewer 2: 这里的锚点丢失了

> 这是一个最基本的 React 应用。用于让您第一次品尝 React，**编辑下面的代码** 然后让它显示您的名字：

Reviewer: ```suggestion
这是一个最基本的 React 应用。要获得React的初体验，**编辑下面的代码** 然后让它显示您的名字：
```

Reviewer 3: 这是一个最基本的 React 应用。要获得 React 的初体验，**编辑下面的代码** 然后让它显示您的名字：

---
title: 开始
translator:
 - x-cold

>  - sooia

Author: ```suggestion
 - sooia
---
```

>  - sooia

Author: ```suggestion
 - sooia
---
```

> 

Author: ```suggestion
---

```

>  - sooia（刘俊）

Reviewer 2: 这里只保留 github 的名称，不要多余内容

Author: @sooia 辛苦移除一下？

Author: ```suggestion
 - sooia
```

<Intro>

> 欢迎阅读 React 文档! 此文档是您在本站点能找到的信息的概览。

Reviewer 4: ```suggestion
欢迎阅读 React 文档！本文档是你在本站点能找到的信息的概览。
```

请避免半角符号的保留以及“您”的使用。符号要求：

> - 所有半角（英文）符号，请转为使用全角（中文）符号
> - 数字使用半角字符
> - -- 破折号，请使用全角 —— 替代
> - 缩写与原文保持一致

</Intro>

<YouWillLearn>

* [如何安装 React](/learn/installation)
* [如何用 React 的思考方式构建用户界面](/learn/thinking-in-react)
* [如何在屏幕上渲染内容](/learn/describing-the-ui)

> * [如何让您的用户界面响应交互](/learn/adding-interactivity)

Reviewer 4: ```suggestion
* [如何让你的用户界面响应交互](/learn/adding-interactivity)
```

> * [随着您的应用的增长，如何保持逻辑的可维护性](/learn/managing-state)

Reviewer 4: ```suggestion
* [随着应用的增长，如何保持逻辑的可维护性](/learn/managing-state)
```

* [必要时如何退出 React](/learn/escape-hatches)

</YouWillLearn>

## 介绍 {/*introduction*/}

> 这是一个最基本的 React 应用。要获得React的初体验，**编辑下面的代码** 然后让它显示您的名字：

Reviewer 4: ```suggestion
这是一个最基本的 React 应用。可以 **编辑下面的代码** 并让它显示你的名字，以获得 React 的初体验：
```

---
title: 开始
translator:
 - x-cold

>  - sooia

Reviewer 4: ```suggestion
 - sooia
 - KnowsCount
```

>  - sooia

Reviewer 5: ```suggestion
 - Bin9o
```
我修改了用户名，能否麻烦@x-cold 修改一下，非常抱歉。

---
title: 开始
translator:
 - x-cold
 - sooia

> ---

Author: ```suggestion
```

---
title: 开始
translator:
 - x-cold
 - sooia
---

> ---

Author: ```suggestion
```

</Sandpack>

### React 是什么？

React 是一个用于构建用户界面的 JavaScript 库

React 擅长交互设计和编程。**它让您可以将一个复杂的用户界面，分解为可嵌套和可重用的片段，我们称之为[组件](/learn/your-first-component)，它们可以很好的进行组合。** 如果您已经有编程的背景，这可能会让您想起用函数来编程。如果您是一名设计师，它可能会让您联想到图层组合设计。如果您对这两门学科都是新手，那也没关系。很多人都是在接触 React 的过程中才熟悉它们的。使用 React 也许还会让您想起用玩具砖块来建造城堡。有时候，它甚至非常有趣。

> React 没有规定要如何构建整个应用程序。它能帮助您定义和组合组件，但在其他问题中不会妨碍您。这意味您可以选择一个生态系统解决方案来解决路由、样式和数据获取等问题，或者您也可以 [使用一个提供了很多内置功能的框架](/learn/start-a-new-react-project#building-with-react-and-a-framework)

Reviewer: ```suggestion
React 没有规定要如何构建整个应用程序。它能帮助您定义和组合组件，但在其他问题中不会妨碍您。这意味您可以选择一个生态系统解决方案来解决路由、样式和数据获取等问题，或者您也可以 [使用一个提供了很多内置功能的框架](/learn/start-a-new-react-project#building-with-react-and-a-framework)。
```

</Sandpack>

### React 是什么？{/*what-is-react*/}

React 是一个用于构建用户界面的 JavaScript 库。

> React 擅长交互设计和编程。**它让您可以将一个复杂的用户界面，分解为可嵌套和可重用的片段，我们称之为[组件](/learn/your-first-component)，它们可以很好的进行组合。** 如果您已经有编程的背景，这可能会让您想起用函数来编程。如果您是一名设计师，它可能会让您联想到图层组合设计。如果您对这两门学科都是新手，那也没关系。很多人都是在接触 React 的过程中才熟悉它们的。使用 React 也许还会让您想起用玩具砖块来建造城堡。有时候，它甚至非常有趣。

Reviewer 4: ```suggestion
React 擅长交互设计和编程。**它让你可以将一个复杂的用户界面，分解为我们称之为 [组件](/learn/your-first-component) 的可嵌套和可重用的片段，而这些组件可以很好的进行组合。** 如果你已经有编程的背景，这可能会让你想起用函数来编程。而如果你是一名设计师，它可能会让你联想到图层组合设计。如果你对这两个领域都不熟悉，那也没关系。很多人都是在接触 React 的过程中才熟悉它们的。使用 React 也许还会让你想起用玩具砖块来建造城堡。有时候，它甚至非常有趣。
```

> React 没有规定要如何构建整个应用程序。它能帮助您定义和组合组件，但在其他问题中不会妨碍您。这意味您可以选择一个生态系统解决方案来解决路由、样式和数据获取等问题，或者您也可以 [使用一个提供了很多内置功能的框架](/learn/start-a-new-react-project#building-with-react-and-a-framework)。

Reviewer 4: ```suggestion
React 没有规定要如何构建整个应用程序。它能帮助你定义和组合组件，但在其他问题中不会妨碍你。这意味你可以选择一个生态系统解决方案来解决路由、样式和数据获取等问题，或者你也可以 [使用一个提供了很多内置功能的框架](/learn/start-a-new-react-project#building-with-react-and-a-framework)。
```

> ### 您可以用 React 做什么？{/*what-can-you-do-with-react*/}

Reviewer 4: ```suggestion
### 你可以用 React 做什么？{/*what-can-you-do-with-react*/}
```

> 实际上，大量开发者们使用 React 来创建各种用户界面——从像按钮和下拉菜单这样的小控件到整个应用程序。 **这些文档将教您在 web 平台上使用 React。** 然而，您在这里学到的大部分内容同样适用于 [React Native](https://reactnative.dev/)，它可以让您为 Android、iOS，甚至 [Windows 和 macOS](https://microsoft.github.io/react-native-windows/) 构建应用程序。

Reviewer 4: ```suggestion
React 能派上一大堆用场！开发者们使用 React 来创建各种用户界面——从像按钮和下拉菜单这样的小控件到整个应用程序。 **这些文档将教你在 web 平台上使用 React。** 然而，你在这里学到的大部分内容同样适用于 [React Native](https://reactnative.dev/)，它可以让你为 Android、iOS，甚至 [Windows 和 macOS](https://microsoft.github.io/react-native-windows/) 构建应用程序。
```

> 如果您好奇您日常使用的哪些产品是用 React 构建的，您可以安装 [React Developer Tools](/learn/react-developer-tools)。 每当您访问使用 React 构建的应用程序或网站时（就像这个文档网站！），它的图标会在工具栏中亮起。

Reviewer 4: ```suggestion
如果你好奇你日常使用的哪些产品是用 React 构建的，你可以安装 [React Developer Tools](/learn/react-developer-tools)。 每当你访问使用 React 构建的应用程序或网站时（比如说这个文档！），它的图标就会在工具栏中亮起来。
```

### React 使用 JavaScript {/*react-uses-javascript*/}

> 使用 React，您将在 JavaScript 中可视化地描述您的逻辑。这需要一些练习。如果您同时学习 JavaScript 和 React，很多人都这么干，但有时，它会更具挑战性！ 从好的方面来说，**大部分学习 React 都是在学习 JavaScript，** 这意味着您将学到的东西远远超出 React。

Reviewer 4: ```suggestion
在 React 中，你会用 JavaScript 来可视化地描述你的逻辑。这需要一定的练习。如果你同时学习 JavaScript 和 React——很多人都这么干——但有时，它会更具挑战！不过还好，**大部分 React 的学习其实都是 JavaScript 的学习，** 也就是说你将学到远远超出 React 的内容。
```

> 使用 [此 JavaScript 概述](https://developer.mozilla.org/zh-CN/docs/Web/JavaScript/A_re-introduction_to_JavaScript) 检查您的知识水平。这将花费您 30 分钟到一个小时，但您会更有信心学习 React。 [MDN](https://developer.mozilla.org/zh-CN/docs/Web/JavaScript) 和 [javascript.info](https://zh.javascript.info/) 是两个很好的参考资源。

Reviewer 4: ```suggestion
使用 [此 JavaScript 概述](https://developer.mozilla.org/zh-CN/docs/Web/JavaScript/A_re-introduction_to_JavaScript) 检查你的知识水平。这将花费你 30 分钟到一个小时，但你会更有信心学习 React。 [MDN](https://developer.mozilla.org/zh-CN/docs/Web/JavaScript) 和 [javascript.info](https://zh.javascript.info/) 是两个很好的参考资源。
```

<DeepDive title="安装 (可选)">

> 如果您刚开始学习 React，您不需要安装任何东西。相反，我们建议您继续使用本网站中出现的 CodeSandbox。 它们看起来像这样：

Reviewer 4: ```suggestion
如果你刚开始学习 React，你不需要安装任何东西。相反，我们建议你继续使用本网站中出现的 CodeSandbox。 它们看起来像这样：
```

</Sandpack>

> 当您准备好开始一个项目时，有多种选择。您可以在线编写 React 代码并将您的代码保存在类似 [CodeSandbox](https://react.new/) 的环境中。您也可以将 React 作为 `<script>` 标签添加到任何 HTML 文件中，以便于在现有网页的一小部分中试用它。或者您可以创建一个全新的 React 应用程序，有或没有框架都行。**使用 [安装](/learn/installation) 页面选择最适合您的方法——但您现在可以跳过它。**

Reviewer 4: ```suggestion
当你准备好开始一个项目时，有多种选择。你可以在线编写 React 代码并将你的代码保存在类似 [CodeSandbox](https://react.new/) 的环境中。你也可以将 React 作为 `<script>` 标签添加到任何 HTML 文件中，以便于在现有网页的一小部分中试用它。或者你可以创建一个全新的 React 应用程序，有或没有框架都行。**请查阅 [安装](/learn/installation) 页面选择最适合你的方法——但现在你可以跳过它。**
```

当您准备好开始一个项目时，有多种选择。您可以在线编写 React 代码并将您的代码保存在类似 [CodeSandbox](https://react.new/) 的环境中。您也可以将 React 作为 `<script>` 标签添加到任何 HTML 文件中，以便于在现有网页的一小部分中试用它。或者您可以创建一个全新的 React 应用程序，有或没有框架都行。**使用 [安装](/learn/installation) 页面选择最适合您的方法——但您现在可以跳过它。**

</DeepDive>

## Learn React {/*learn-react*/}
## 学习 React {/*learn-react*/}

There are a few ways to get started:
有几种方法可以开始：

- If you're **feeling impatient and learn by example,** head straight to **[Thinking in React](/learn/thinking-in-react)**. This tutorial doesn't explain the syntax in detail, but it will give you an idea of what it feels like to build user interfaces with React.
- If you're **familiar with the concepts and want to browse the available APIs**, check out **[the API reference](/reference)**.

> - The rest of this documentation is organized in chapters that **introduce each concept step by step**--with many interactive examples, detailed explanations, and challenges to check your understanding. You don't have to read them sequentially, but each next page assumes you're familiar with concepts from the previous pages.

Reviewer 4: 这里行数有问题。

> - 如果您**觉得迫不及待并通过例子学习**，直接前往 **[React 哲学](/learn/thinking-in-react)**。 该教程不会详细解释语法，但它会让您了解使用 React 构建用户界面的感受。

Reviewer 4: ```suggestion
- 如果你**迫不及待并适合通过例子学习**，直接前往 **[React 哲学](/learn/thinking-in-react)**。该教程不会详细解释语法，但它会让你了解使用 React 构建用户界面的感受。
```

> - 如果您**熟悉这些概念并希望浏览可用的 API**，请查看 **[API 参考](/reference)**。

Reviewer 4: ```suggestion
- 如果你 **熟悉这些概念并希望浏览可用的 API**，请查看 **[API 参考](/reference)**。
```

</Sandpack>

当你准备好开始一个项目时，有多种选择。你可以在线编写 React 代码并将你的代码保存在类似 [CodeSandbox](https://react.new/) 的环境中。你也可以将 React 作为 `<script>` 标签添加到任何 HTML 文件中，以便于在现有网页的一小部分中试用它。或者你可以创建一个全新的 React 应用程序，有或没有框架都行。**请查阅 [安装](/learn/installation) 页面选择最适合你的方法——但现在你可以跳过它。**

</DeepDive>

## 学习 React {/*learn-react*/}

有几种方法可以开始：

- 如果你 **迫不及待并适合通过例子学习**，直接前往 **[React 哲学](/learn/thinking-in-react)**。该教程不会详细解释语法，但它会让你了解使用 React 构建用户界面的感受。
- 如果你 **熟悉这些概念并希望浏览可用的 API**，请查看 **[API 参考](/reference)**。

> - 本文档的其余部分按章节组织，**逐步介绍每个概念**--带有许多交互式例子、详细说明和挑战以保证您的理解。 您不必按顺序阅读它们，但每个下一页都假设您熟悉前几页中的概念。

Reviewer 4: ```suggestion
- 本文档的其余部分按章节组织，**逐步介绍每个概念** —— 带有许多交互式例子、详细说明和挑战以保证你的理解。 你不一定要按顺序阅读它们，但是每个主题都需要你熟悉前几页中的概念。
```

> 为了节省您的时间，我们在下面提供**每章的简要概述**。 

Reviewer 4: ```suggestion
为了节省您的时间，我们在下面提供 **每章的简要概述**。 
```

### 第 1 章概述：描述 UI {/*chapter-1-overview-describing-the-ui*/}

> React 应用程序由被称为 ["components"](/learn/your-first-component) 的独立 UI 部分构建。 React 组件是一个 JavaScript 函数，您可以在其中添加标记。 组件可以小到一个按钮，也可以大到整个页面。下面的例子中，一个*父*组件 `Gallery` 渲染了三个*子*组件 `Profile` ：

Reviewer 4: ```suggestion
React 应用程序由被称为 ["组件"](/learn/your-first-component) 的独立 UI 部分构建。 React 组件是一个 JavaScript 函数，您可以在其中添加标记。 组件可以小到一个按钮，也可以大到整个页面。下面的例子中，一个 *父* 组件 `Gallery` 渲染了三个 *子* 组件 `Profile` ：
```

其它地方都翻译了出来，这里不翻译有些奇怪吧。

</Sandpack>

> 上面例子中的标记看起来很像 HTML。 这种语法叫做 [JSX](/learn/writing-markup-with-jsx)，它更严格一些（例如，所有标签必须闭合）。请注意，CSS 类在 JSX 中指定为 `className`。

Reviewer 4: ```suggestion
上面例子中的标记看起来很像 HTML。这种语法叫做 [JSX](/learn/writing-markup-with-jsx)，它更严格一些（例如，所有标签必须闭合）。请注意，CSS 类在 JSX 中指定为 `className`。
```

> 就像你可以将一些信息传递给浏览器`<img>` 标签一样，你也可以将信息传递给你自己的组件，比如`<Profile>`。 此类信息称为 [_props_](/learn/passing-props-to-a-component)。 下面的例子中，三个 `<Profile>` 接收不同的 props：

Reviewer 4: ```suggestion
就像你可以将一些信息传递给浏览器`<img>` 标签一样，你也可以将信息传递给你自己的组件，比如`<Profile>`。 此类信息称为 [_props_](/learn/passing-props-to-a-component)。下面的例子中，三个 `<Profile>` 接收不同的 props：
```

</Sandpack>

> 你可能想知道为什么 `className="avatar"` 使用引号，而 `src={imageUrl}` 使用花括号。在 JSX 中，花括号就像一个 ["进入 JavaScript 的窗口"](/learn/javascript-in-jsx-with-curly-braces)。 它们让您可以在标记中直接运行一些 JavaScript！所以 `src={imageUrl}` 读取第一行声明的 `imageUrl` prop，并从父组件 `Gallery` 传递过来。

Reviewer 4: ```suggestion
你可能想知道为什么 `className="avatar"` 使用引号，而 `src={imageUrl}` 使用花括号。在 JSX 中，花括号就像一个 ["进入 JavaScript 的窗口"](/learn/javascript-in-jsx-with-curly-braces)。 它们让你可以在标记中直接运行一些 JavaScript！所以 `src={imageUrl}` 读取第一行声明的 `imageUrl` prop，并从父组件 `Gallery` 传递过来。
```

> 在上面的例子中，所有的数据都是直接写在标记中的。但是，您通常希望将其单独保存。下面的例子，数据保存在一个数组中。在 React 中，你使用 JavaScript 函数，如 [`map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) 来[渲染列表](/learn/rendering-lists) 的东西。 

Reviewer 4: ```suggestion
在上面的例子中，所有的数据都是直接写在标记中的。但是，很多情况下我们想将其分开存储。下面的例子，数据保存在一个数组中。在 React 中，一般使用例如 [`map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map)  的 JavaScript 函数，来 [渲染一个列表](/learn/rendering-lists) 的东西。 
```

<LearnMore path="/learn/describing-the-ui">

> 阅读**[描述 UI](/learn/describing-the-ui)** 了解如何让东西出现在屏幕上，包括声明组件、导入它们、用大括号编写 JSX 以及编写条件和列表。

Reviewer 4: ```suggestion
阅读 **[描述 UI](/learn/describing-the-ui)** 了解如何让东西出现在屏幕上，引用声明组件、导入它们、用大括号编写 JSX 以及编写条件和列表。
```

</LearnMore>

### 第 2 章概述：添加交互性 {/*chapter-2-overview-adding-interactivity*/}

作为交互的结果，组件通常需要更改屏幕上的内容。输入表单应该更新输入字段，单击图像轮播上的“下一步”应该更改显示的图像，单击“购买”将产品放入购物车。组件需要“记住”一些东西：当前输入值、当前图像、购物车。在 React 中，这种特定于组件的内存称为 [*state*](/learn/state-a-components-memory)。

> 您可以使用 [`useState`](/reference/usestate) Hook 向组件添加 state。 Hooks 是一种特殊的函数，可以让你的组件使用 React 特性（state 是这些特性之一）。`useState` Hook 允许你声明一个 state 变量。它接受初始 state 并返回一对值：当前 state 和一个允许您更新它的 state 设置器函数。

Reviewer 4: ```suggestion
你可以使用 [`useState`](/reference/usestate) Hook 向组件添加 state。 Hooks 是一种特殊的函数，可以让你的组件使用 React 特性（state 是这些特性之一）。`useState` Hook 允许你声明一个 state 变量。它接受初始 state 并返回一对值：当前 state 以及一个允许你更新它的 state 设置器函数。
```

  return (
    <>
      <button onClick={handleNextClick}>
        下一个
      </button>
      <h2>

>         {sculpture.artist} 的 <i>{sculpture.name} </i>

Author: ```suggestion
        <i>{sculpture.name} </i>
        {sculpture.artist} 的 <i>{sculpture.name} </i>
```

}, {
  name: 'Terracotta Army',
  artist: 'Unknown Artist',

>   description: 'The Terracotta Army is a collection of terracotta sculptures depicting the armies of Qin Shi Huang, the first Emperor of China. The army consited of more than 8,000 soldiers, 130 chariots with 520 horses, and 150 cavalry horses.',

Reviewer 4: ```suggestion
  description: 'The Terracotta Army is a collection of terracotta sculptures depicting the armies of Qin Shi Huang, the first Emperor of China. The army consisted of more than 8,000 soldiers, 130 chariots with 520 horses, and 150 cavalry horses.',
```

</Sandpack>

> State 也可以保存复杂的值。例如，如果您正在实现一个表单，您可以在 state 中保持一个具有不同字段的对象。 下面示例中的 `...` 语法允许您[基于现有对象创建新对象](/learn/updating-objects-in-state)。

Reviewer 4: ```suggestion
State 也可以保存复杂的值。例如，如果你正在实现一个表单，你就可以在 state 中保持一个具有不同字段的对象。 下面示例中的 `...` 语法允许你 [基于现有对象创建新对象](/learn/updating-objects-in-state)。
```

</Sandpack>

> 您还可以将数组保持在 state。 这允许您添加、删除或更改列表中的内容以响应用户交互。根据您想要做什么，这里有[从现有数组创建新数组的不同方法](/learn/updating-arrays-in-state)。

Reviewer 4: ```suggestion
你还可以将数组保持在 state。 这允许你添加、删除或更改列表中的内容以响应用户交互。按照你的需求，有不同的 [从现有数组创建新数组的方法](/learn/updating-arrays-in-state)。
```

<LearnMore path="/learn/adding-interactivity">

> 阅读**[添加交互性](/learn/adding-interactivity)** 了解如何在交互时更新屏幕，包括添加事件处理程序、声明和更新 state，以及在 state 中更新对象和数组的不同方法。

Reviewer 4: ```suggestion
阅读* *[添加交互性](/learn/adding-interactivity)** 了解如何在交互时更新屏幕，包括添加事件处理程序、声明和更新 state，以及在 state 中更新对象和数组的不同方法。
```

</LearnMore>

### 第 3 章概述： 管理 State {/*chapter-3-overview-managing-state*/}

> 您经常会面临选择_究竟要把什么_放进 state 的问题。 您应该使用一个 state 变量还是多个 state 变量？ 对象还是数组？ 你应该如何[构建你的 state](/learn/choosing-the-state-structure)？ 最重要的原则是**避免冗余 state**。 如果某些信息永远不会改变，它就不应该处于 state。 如果某些信息是通过 props 从父组件收到的，它不应该处于 state 中。 如果某些信息可以从其他 props 或 state 计算得出，它也不应该处于 state 中！

Reviewer 4: ```suggestion
你经常会面临选择 _究竟要把什么_ 放进 state 的问题。 你应该使用一个 state 变量还是多个 state 变量？ 对象还是数组？ 你应该如何 [构建你的 state](/learn/choosing-the-state-structure)？ 最重要的原则是 **避免冗余 state**。 如果某些信息永远不会改变，它就不应该处于 state。 如果某些信息是通过 props 从父组件收到的，它不应该处于 state 中。 如果某些信息可以从其他 props 或 state 计算得出，它也不应该处于 state 中！
```

  return (
    <>

>       <label>

Author: ```suggestion
      <h2>Let’s check you in</h2>
      <label>
```

    <>
      <h2>Let’s check you in</h2>
      <label>
        名：{' '}
        <input
          value={firstName}
          onChange={handleFirstNameChange}
        />
      </label>
      <label>
        姓：{' '}
        <input
          value={lastName}
          onChange={handleLastNameChange}
        />
      </label>

>       <h3>

Author: ```suggestion
      <p>
```

</Sandpack>

> 您可以通过在组件渲染时计算 `fullName` 来删除它并简化代码：

Reviewer 4: ```suggestion
你可以通过在组件渲染时计算 `fullName` 来删除它并简化代码：
```

  return (
    <>

>       <label>

Author: ```suggestion
      <h2>Let’s check you in</h2>
      <label>
```

    <>
      <h2>Let’s check you in</h2>
      <label>
        名：{' '}
        <input
          value={firstName}
          onChange={handleFirstNameChange}
        />
      </label>
      <label>
        姓：{' '}
        <input
          value={lastName}
          onChange={handleLastNameChange}
        />
      </label>

>       <h3>

Author: ```suggestion
      <p>
```

>       <h3>

Author: ```suggestion
      <p>
```

        您的姓名是：{fullName}

>       </h3>

Author: ```suggestion
      </p>
```

</Sandpack>

> 有时，您希望两个组件的 state 始终一起更改。要做到这一点，在它们两个中都删除该 state，将它移动到它们最近的公共父节点，然后通过 props 将它传递给它们。 这被称为["提升 state"](/learn/sharing-state-between-components)，这是编写 React 代码时最常见的事情之一。 例如，在像下面这样的手风琴组件中，一次应该只有一个面板处于激活状态。父组件不应该将激活 state 保存在每个单独的面板内，而是保存在父组件的 state 中并通过 props 指定给子组件。

Author: ```suggestion
这可能看起来像一个小变化，但在 React 应用中很多 bug 都是这么修复的。

有时，您希望两个组件的 state 始终一起更改。要做到这一点，在它们两个中都删除该 state，将它移动到它们最近的公共父节点，然后通过 props 将它传递给它们。 这被称为["提升 state"](/learn/sharing-state-between-components)，这是编写 React 代码时最常见的事情之一。 例如，在像下面这样的手风琴组件中，一次应该只有一个面板处于激活状态。父组件不应该将激活 state 保存在每个单独的面板内，而是保存在父组件的 state 中并通过 props 指定给子组件。
```

</Sandpack>

这可能看起来像一个小变化，但在 React 应用中很多 bug 都是这么修复的。

> 有时，您希望两个组件的 state 始终一起更改。要做到这一点，在它们两个中都删除该 state，将它移动到它们最近的公共父节点，然后通过 props 将它传递给它们。 这被称为["提升 state"](/learn/sharing-state-between-components)，这是编写 React 代码时最常见的事情之一。 例如，在像下面这样的手风琴组件中，一次应该只有一个面板处于激活状态。父组件不应该将激活 state 保存在每个单独的面板内，而是保存在父组件的 state 中并通过 props 指定给子组件。

Reviewer 4: ```suggestion
你有的时候可能需要两个组件的 state 同时更改。要做到这一点，在它们两个中都删除该 state，将它移动到它们最近的公共父节点，然后通过 props 将它传递给它们。 这被称为 ["提升 state"](/learn/sharing-state-between-components)，这是编写 React 代码时最常见的事情之一。 例如，在像下面这样的手风琴组件中，一次应该只有一个面板处于激活状态。父组件不应该将激活 state 保存在每个单独的面板内，而是保存在父组件的 state 中并通过 props 指定给子组件。
```

  const [activeIndex, setActiveIndex] = useState(0);
  return (
    <>
      <h2>阿拉木图, 哈萨克斯坦</h2>
      <Panel
        title="关于"
        isActive={activeIndex === 0}
        onShow={() => setActiveIndex(0)}
      >
        阿拉木图有大约 200 万人口，是哈萨克斯坦最大的城市。从 1929 年到 1997 年，它是哈萨克斯坦的首都。
      </Panel>
      <Panel
        title="词源"
        isActive={activeIndex === 1}
        onShow={() => setActiveIndex(1)}
      >

>         这个名字源自 <span lang="kk-KZ">алма</span>，哈萨克语中的“苹果”一词，通常被翻译为“苹果之城”。 事实上，阿拉木图周边地区被认为是苹果的祖先家园，而野生<i lang="la">Malus Siversii (天山野苹果)</i> 可能是现代家养苹果祖先。

Reviewer 4: ```suggestion
        这个名字源自 <span lang="kk-KZ">алма</span>，哈萨克语中的“苹果”一词，通常被翻译为“苹果之城”。事实上，阿拉木图周边地区被认为是苹果的祖先家园，而野生<i lang="la">Malus Siversii (天山野苹果)</i> 可能是现代家养苹果祖先。
```

<LearnMore path="/learn/managing-state">

阅读 **[管理 State](/learn/managing-state)** 了解如何保持组件的可维护性，包括如何很好地构建 state，如何在组件之间共享 state，以及如何将 state 传递到树的深处。

</LearnMore>

## 下一步 {/*next-steps*/}

> 这个页面是快节奏的！如果读到这里，您已经看到了日常使用 React 的 80%。

Reviewer 4: ```suggestion
这一章节奏很快！如果已经读到这里，你就已经读了 React 80% 的日常使用内容。
```

> 您接下来的步骤取决于您想做什么：

Reviewer 4: ```suggestion
接下来的步骤取决于你：
```

> * 如果您想在本地创建一个 React 项目，请转到 [安装](/learn/installation)。

Reviewer 4: ```suggestion
* 如果你想在本地创建一个 React 项目，请转到 [安装](/learn/installation)。
```

> * 如果您想了解在 React 中构建 UI 在实践中是什么感觉，请阅读 [React 哲学](/learn/thinking-in-react)。

Reviewer 4: ```suggestion
* 如果你想了解在 React 中构建 UI 在实践中是什么感觉，请阅读 [React 哲学](/learn/thinking-in-react)。
```

* 或者，从 [描述 UI](/learn/describing-the-ui) 开始，仔细阅读第一章。

> 当您需要了解 API 必要的细节时，不要忘记阅读 [API 参考](/reference)！

Reviewer 4: ```suggestion
当你需要了解 API 必要的细节时，不要忘记阅读 [API 参考](/reference)！
```
